### PR TITLE
[Python] Operations pages batch 2 [Replace C# samples]

### PR DESCRIPTION
**Pull request by reebhub**
Related issues:
[RDoc-2784](https://issues.hibernatingrhinos.com/issue/RDoc-2784/Python-Client-API-Operations-Maintenance-Indexes-Put-indexes-Replace-C-samples) Put indexes
[RDoc-2785](https://issues.hibernatingrhinos.com/issue/RDoc-2785/Python-Client-API-Operations-Maintenance-Indexes-Resume-indexing-Replace-C-samples) Resume indexing
[RDoc-2786](https://issues.hibernatingrhinos.com/issue/RDoc-2786/Python-Client-API-Operations-Maintenance-Indexes-Pause-indexing-Replace-C-samples) Pause indexing
[RDoc-2791](https://issues.hibernatingrhinos.com/issue/RDoc-2791/Python-Client-API-Operations-Maintenance-Indexes-Resume-index-Replace-C-samples) Resume index
[RDoc-2792](https://issues.hibernatingrhinos.com/issue/RDoc-2792/Python-Client-API-Operations-Maintenance-Indexes-Pause-index-Replace-C-samples) Pause index
[RDoc-2793](https://issues.hibernatingrhinos.com/issue/RDoc-2793/Python-Client-API-Operations-Maintenance-Indexes-Delete-index-Replace-C-samples) Delete index
[RDoc-2794](https://issues.hibernatingrhinos.com/issue/RDoc-2794/Python-Client-API-Operations-Maintenance-Indexes-Delete-index-errors-Replace-C-samples) Delete index errors
[RDoc-2795](https://issues.hibernatingrhinos.com/issue/RDoc-2795/Python-Client-API-Operations-Maintenance-Indexes-Disable-index-Replace-C-samples) Disable index
[RDoc-2796](https://issues.hibernatingrhinos.com/issue/RDoc-2796/Python-Client-API-Operations-Maintenance-Indexes-Enable-index-Replace-C-samples) Enable index
[RDoc-2797](https://issues.hibernatingrhinos.com/issue/RDoc-2797/Python-Client-API-Operations-Maintenance-Indexes-Reset-index-Replace-C-samples) Reset index
[RDoc-2798](https://issues.hibernatingrhinos.com/issue/RDoc-2798/Python-Client-API-Operations-Maintenance-Indexes-Set-index-lock-mode-Replace-C-samples) Set index lock mode

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+| Parameters | Type | Description |
+| - |- | - |
+| **\*indexes_to_add** | `IndexDefinition` | Definitions of indexes to deploy |
+
+<a id="indexDefinition" />
+
+| `IndexDefinition` parameter| Type | Description |
+| - |- | - |
+| **name** | `str` | Name of the index, a unique identifier |
+| **maps** | `HashSet[str]` | All the map functions for the index |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:79]:
```suggestion
| **maps** | `Set[str]` | All the map functions for the index |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+| **reduce** | `str` | The index reduce function |
+| **deployment_mode** | `IndexDeploymentMode` | Deployment mode<br>(PARALLEL, ROLLING) |
+| **state** | `IndexState` | State of index<br>(NORMAL, DISABLED, IDLE, ERROR) |
+| **priority** | `IndexPriority` | Priority of index<br>(LOW, NORMAL, HIGH) |
+| **lock_mode** | `IndexLockMode` | Lock mode of index<br>(UNLOCK, LOCKED_IGNORE, LOCKED_ERROR) |
+| **fields** | `Dictionary[str, IndexFieldOptions]` | _IndexFieldOptions_ per index field |
+| **additional_sources** | `Dictionary[str, str]` | Additional code files to be compiled with this index |
+| **additional_assemblies** | `HashSet[AdditionalAssembly]` | Additional assemblies that are referenced |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:87]:
```suggestion
| **additional_assemblies** | `Set[AdditionalAssembly]` | Additional assemblies that are referenced |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+| - |- | - |
+| **name** | `str` | Name of the index, a unique identifier |
+| **maps** | `HashSet[str]` | All the map functions for the index |
+| **reduce** | `str` | The index reduce function |
+| **deployment_mode** | `IndexDeploymentMode` | Deployment mode<br>(PARALLEL, ROLLING) |
+| **state** | `IndexState` | State of index<br>(NORMAL, DISABLED, IDLE, ERROR) |
+| **priority** | `IndexPriority` | Priority of index<br>(LOW, NORMAL, HIGH) |
+| **lock_mode** | `IndexLockMode` | Lock mode of index<br>(UNLOCK, LOCKED_IGNORE, LOCKED_ERROR) |
+| **fields** | `Dictionary[str, IndexFieldOptions]` | _IndexFieldOptions_ per index field |
+| **additional_sources** | `Dictionary[str, str]` | Additional code files to be compiled with this index |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:86]:
```suggestion
| **additional_sources** | `Dict[str, str]` | Additional code files to be compiled with this index |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+| `IndexDefinition` parameter| Type | Description |
+| - |- | - |
+| **name** | `str` | Name of the index, a unique identifier |
+| **maps** | `HashSet[str]` | All the map functions for the index |
+| **reduce** | `str` | The index reduce function |
+| **deployment_mode** | `IndexDeploymentMode` | Deployment mode<br>(PARALLEL, ROLLING) |
+| **state** | `IndexState` | State of index<br>(NORMAL, DISABLED, IDLE, ERROR) |
+| **priority** | `IndexPriority` | Priority of index<br>(LOW, NORMAL, HIGH) |
+| **lock_mode** | `IndexLockMode` | Lock mode of index<br>(UNLOCK, LOCKED_IGNORE, LOCKED_ERROR) |
+| **fields** | `Dictionary[str, IndexFieldOptions]` | _IndexFieldOptions_ per index field |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:85]:
```suggestion
| **fields** | `Dict[str, IndexFieldOptions]` | _IndexFieldOptions_ per index field |
```

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+| **additional_assemblies** | `HashSet[AdditionalAssembly]` | Additional assemblies that are referenced |
+| **configuration** | `IndexConfiguration` | Can override [indexing configuration](../../../../server/configuration/indexing-configuration) by setting this dictionary |
+| **output_reduce_to_collection** | `str` | A collection name for saving the reduce results as documents |
+| **reduce_output_index** | `int` | This number will be part of the reduce results documents IDs |
+| **pattern_for_output_reduce_to_collection_references** | `str` | Pattern for documents IDs which reference IDs of reduce results documents |
+| **pattern_references_collection_name** | `str` | A collection name for the reference documents created based on provided pattern |
+
+| `store.maintenance.send(put_indexes_op)` return value | Description |
+| - |- |
+| `PutIndexResult[]` | List of _PutIndexResult_ per index |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:96]:
```suggestion
| `List[PutIndexResult]` | List of _PutIndexResult_ per index |
```

**Review by poissoncorp — APPROVED** [review]

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:79]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:87]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:86]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:85]: done

File: Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/client-api/operations/maintenance/indexes/put-indexes.python.markdown:96]: done

**Pull request merged**